File: day01/code_01b.py
with open("input_01.txt") as f:
    content = f.readlines()

content = [int(x.strip()) for x in content]

len(content)

sumvar = 0
mylist = [sumvar]
for i in range(0,len(content)):
    sumvar = sumvar + content[i]
    if sumvar in mylist:
        print(sumvar)
    mylist = mylist + [sumvar]

len(mylist)

import pandas as pd

df = pd.DataFrame(mylist)

df.columns = ["vals"]

for i in range(0,len(mylist)):
    for j in range(i+1,len(mylist)):
        if mylist[i]==mylist[j]:
            print(mylist[i])

len(mylist)
df.vals.value_counts()

content
mylist

File: day01/code_01b.cs
using System;
using System.IO;
using System.Collections.Generic;

public class Test
{
  static void Main ()
  {
    int sumVar = 0;
    List<int> myList = new List<int>();
    myList.Add(sumVar);

    foreach (string line in File.ReadLines("input_01.txt"))
    {
      int numVal = Int32.Parse(line);
      sumVar = sumVar + numVal;
      myList.Add((int)sumVar);
    }

    bool notYet = true;
    for (int i = 0; i < myList.Count; i++)
    {
      for (int j = i+1; j < myList.Count; j++)
      {
        if (myList[i] == myList[j] && notYet)
        {
          Console.WriteLine (myList[i]);
          notYet = false;
        }
      }
    }
  }
}
